prow/manifests/overlays/metal3/config.yaml

```diff
@@ -2602,29 +2603,19 @@ presubmits:
           value: "TRUE"
         image: docker.io/pipelinecomponents/markdownlint-cli2:0.9.0@sha256:71370df6c967bae548b0bfd0ae313ddf44bfad87da76f88180eff55c6264098c
         imagePullPolicy: Always
-  - name: markdownlint
```

Comment:
I think the plan was to support 2-3 releases of ironic-image (roughly 6-8 months coverage), we're at 4 already
I'm going to leave it there for now but we should probably discuss about this

Reply:
Yeah, it is a bit problematic as they aren't released in the same time, so we probably end up in a situation where we support something like:
- 24.0 (tested)
- 24.1 (supported)
- 25.0 (not used by Metal3) 
- 25.1 (supported)
- main (development)

That'd would happen if we'd skip 25.0 for upcoming 1.8 series due 25.1 being out already. (not going to happen this time, but this will happen in future)... then the support would be dropped for 25.0 which of course would be not logical from ironic-image perspective.... 

We know there are compatibility issues between ironic-image releases so we also can't (always) be upgrading ironic-image in older minor releases, it kind of is tied to BMO version.

Soo, maybe we can live with this until ironic-standalone-operator delivers salvation? :)

Reply:
So as far as I can see this (and we had a long discussion the 2024.06.12 community meeting). If CAPM3 1.5 was released and was/is tested with a BMO that uses 23.1 Ironic image then even after 23.1 Ironic source branch (not the ironic-image) looses support the CAPM3 1.5 is still tested against the 23.1, CAPM3 and BMO would continue using the an be tested against the Ironic version they were released with, even if the Ironic source they use is not supported anymore.

Ofc there could situations arise that we have to handle, IMO we should handle them like this:

- In case after an Ironic source branch looses it's upstream support but BMO and CAPM3 still rely on said branch and we would need a fix to the image (not the ironic source) we would just release an Ironic-image patch release e.g. 23.1.1. and probably we would need to release a patch release of CAPM3 and BMO that would use the new image.

- If we would need to backport something to an unsupported Ironic source branch not just to the image (and this is very  very unlikely but anything could happen) then we have to consider releasing a BMO and CAPM3 and release that rely on a new release of ironic-image that uses a different ironic source branch if possible on a stable branch but not exclusively, it could be bugfix branch or Nordix custom branch or a random commit from Ironic main. In this last scenario I don't think we have an elegant solution we just have to keep the CAPM3 and BMO version supported at any cost.

During our discussion related to this topic we also come to the conclusion that IPA has to be pinned because right now even a 23.1 ironic image is used with the latest IPA and that will break at some point so we have discussed with @elfosardo that we will look into possibilities of where can we build and store IPA releases and how we should release the IPA downloader.